client/components/infinite-list/index.jsx

```diff
@@ -154,9 +159,12 @@ export default class InfiniteList extends React.Component {
 		// we may have guessed item heights wrong - now we have real heights
 		if ( ! this.isScrolling ) {
 			this.cancelAnimationFrame();
-			this.updateScroll( {
-				triggeredByScroll: false,
-			} );
+			// updateScroll misbehaves when it's called syncronously from componentDidUpdate.
+			setTimeout( () => {
```

Comment:
could use `Promise.resolve().then( ... )` here to schedule the check [even closer](https://stackoverflow.com/questions/38752620/promise-vs-settimeout) to `componentDidUpdate` running.